packages/actions/test/unit/security.test.ts

```diff
@@ -89,70 +110,281 @@ describe("Security", () => {
+            // @todo requires adding the checks to the cloud function
+            it("should prevent a user with a non reputable GitHub account from authenticating to the Firebase", async () => {})
+            /// @note If a coordinator disables an account, this should not be allowed to authenticate
+            it("should prevent a disabled account from loggin in (OAuth2)", async () => {
+                const auth = createOAuthDeviceAuth({
+                    clientType,
+                    clientId,
+                    scopes: ["gist"],
+                    onVerification: simulateOnVerification
+                })
+                const { token } = await auth({
+                    type: tokenType
+                })
+                // Get and exchange credentials.
+                const userFirebaseCredentials = GithubAuthProvider.credential(token)
+                await signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)
+
+                const user = getCurrentFirebaseAuthUser(userApp)
+                userId = user.uid
+                // Disable user.
+                const disabledRecord = await adminAuth.updateUser(user.uid, { disabled: true })
+                expect(disabledRecord.disabled).to.be.true
+
+                await signOut(userAuth)
+
+                await expect(signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)).to.be.rejectedWith(
+                    "Firebase: Error (auth/user-disabled)."
+                )
+            })
+        }
+        /// @note It should not be possible to enumerate valid email addresses
+        /// using the error messages returned by the server (wrong email/ wrong password)
+        /// @todo This feature needs to be enabled in one account,
+        /// document this feature to prevent enumeration attacks
+        it.skip("should not allow a user to enumerate valid emails", async () => {
+            // @link https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection
+            const wrongPassword = "wrongPassword"
+            const wrongEmail = "wrongEmail"
+            expect(signInWithEmailAndPassword(userAuth, wrongEmail, wrongPassword)).to.not.be.rejectedWith(
+                "Firebase: Error (auth/invalid-email)."
+            )
+            expect(signInWithEmailAndPassword(userAuth, users[1].data.email, wrongPassword)).to.not.be.rejectedWith(
+                "Firebase: Error (auth/wrong-password)."
+            )
+        })
+        /// @note Firebase should enforce rate limiting to prevent denial of service or consumption of resources
+        /// @todo check firebase settings/docs
+        it.skip("should rate limit after a large number of failed attempts", async () => {
+            let err: any
+            try {
+                for (let i = 0; i < 1000; i++) {
+                    await expect(signInToFirebaseWithCredentials(userApp, new OAuthCredential())).to.be.rejectedWith(
```

Comment:
*not sure this makes sense to test locally; what happens when you run against prod?

Reply:
oh that's the error you get when providing invalid OAuth2 creds: I believe it says localhost and provider undefined because it's an empty object and cannot deduct any information of the OAuth2 endponits

Reply:
Looks like failed requests (both OAuth2 and email/pwd) do not cause the caller to be rate limited. However, a user that attempts to auth using a valid token/email&pwd many times in a row, will get locked out, which is a valid test case imo - a user might try to take down the authentication service but instead they get locked out. Added those tests